packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -538,9 +566,13 @@ export class BlobManager {
             );
             const blobId = pathParts[2];
 
-            // The unused blobId could be a localId. If so, remove it from the redirect table and continue. The
-            // corresponding storageId may still be used either directly or via other localIds.
-            if (this.redirectTable?.has(blobId)) {
+            // GC tombstones these blobs
+            if (tombstone) {
+                this.tombstonedBlobs.add(blobId);
+                continue;
+            } else if (this.redirectTable?.has(blobId)) {
```

Comment:
This check is not needed actually. You can directly call `this.redirectTable.delete(blobId)`. The code is this way because of some changes that happened with blob manager.